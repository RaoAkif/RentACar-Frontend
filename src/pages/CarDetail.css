.carDetail {
  display: flex;
  justify-content: space-around;
  align-items: flex-end;
}

.carImage {
  position: absolute;
  top: 15%;
  right: 60%;
}

.carInfo {
  position: absolute;
  top: 30%;
  right: 7%;
  display: flex;
  flex-direction: column;
  align-items: flex-end;
}

.carInfo > * {
  padding-bottom: 1rem;
}

.name-model {
  display: flex;
  align-items: center;
  padding: 0;
  padding-top: 0.5rem;
}

p {
  font-size: 1.2rem;
}

.name-model > p {
  padding-left: 1rem;
}

#form-btn-cd {
  background: var(--green-color);
  border: none;
  border-radius: 2rem;
  width: 60%;
  padding: 0.8rem;
  color: white;
  font-size: 1.5rem;
  margin-top: 1rem;
}

.carPrice {
  background-color: rgb(221, 220, 220);
  padding: 0.5rem 6.7rem;
}
