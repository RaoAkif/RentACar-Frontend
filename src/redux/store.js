import { configureStore } from '@reduxjs/toolkit';
import cakeReducer from './cake/cakeSlice';
import icecreamReducer from './icecream/icecreamSlice';
import userReducer from './user/userSlice';

const store = configureStore({
  reducer: {
    cake: cakeReducer,
    icecream: icecreamReducer,
    user: userReducer,
  },
});

export default store;
