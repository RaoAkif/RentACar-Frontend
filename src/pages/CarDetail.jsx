import React, { useEffect } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { useParams, useNavigate } from 'react-router-dom';
import { fetchCarDetails } from '../redux/car/detail';
import './CarDetail.css';

function CarDetail() {
  const dispatch = useDispatch();
  const navigate = useNavigate();
  const { id } = useParams();
  const car = useSelector((store) => store.carDetail);

  useEffect(() => {
    dispatch(fetchCarDetails(id));
  }, [dispatch]);

  const redirectHandler = () => {
    navigate('/add_rental');
  };

  return (
    <div className="carDetail">
      <div className="car-image-div"><img className="car-image" src={car.image} alt={car.name + car.model} /></div>
      <div className="carInfo">
        <div className="name-model">
          <h3>{car.name}</h3>
          <p>{car.model}</p>
        </div>
        <p className="carPrice">{car.rent}</p>
        <p className="carDesc">{car.desc}</p>
        <button id="form-btn-cd" type="button" onClick={redirectHandler}>Rent</button>
      </div>
    </div>
  );
}

export default CarDetail;
