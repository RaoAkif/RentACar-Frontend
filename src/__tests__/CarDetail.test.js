import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import store from '../redux/configureStore';
import { render } from '@testing-library/react';
import CarDetail from '../pages/CarDetail';

describe('CarDetail', () => {  
  it('should render correctly', () => {
    const { container } = render(
      <BrowserRouter>
        <Provider store={store}>
          <CarDetail />
        </Provider>,
      </BrowserRouter>,
    );

    expect(container).toMatchSnapshot();
  });
});
