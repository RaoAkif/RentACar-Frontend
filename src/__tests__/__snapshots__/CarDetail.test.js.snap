// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CarDetail should render correctly 1`] = `
<div>
  <div
    class="car-detail"
  >
    <div
      class="details-image-div"
    >
      <img
        class="details-image"
      />
    </div>
    <div
      class="car-info"
    >
      <h3 />
      <table
        class="table table-striped table-bordered details-table"
      >
        <tbody>
          <tr>
            <td
              class="text-start"
            >
              <span>
                Model:
              </span>
            </td>
            <td
              class="text-end"
            />
          </tr>
          <tr>
            <td
              class="text-start"
            >
              <span>
                Rent:
              </span>
            </td>
            <td
              class="text-end"
            />
          </tr>
        </tbody>
      </table>
      <p
        class="car-desc"
      />
      <button
        class="btn btn-green"
        id="form-btn-cd"
        type="button"
      >
        Rent
      </button>
    </div>
  </div>
  ,
</div>
`;
